File: src/main/java/rosatech/client/renderer/textures/RosaTextures.java
package rosatech.client.renderer.textures;

import gregtech.client.renderer.texture.cube.OrientedOverlayRenderer;
import gregtech.client.renderer.texture.cube.SimpleOverlayRenderer;
import gregtech.client.renderer.texture.cube.SimpleSidedCubeRenderer;


public class RosaTextures {

    public static SimpleSidedCubeRenderer MAGICAL_CASING;
    public static SimpleOverlayRenderer MANA_HATCH_INPUT_OVERLAY;
    public static SimpleOverlayRenderer MANA_HATCH_OUTPUT_OVERLAY;
    public static SimpleOverlayRenderer LIFE_ESSENCE_INPUT_OVERLAY;
    public static SimpleOverlayRenderer LIFE_ESSENCE_OUTPUT_OVERLAY;


    public static void preInit() {
        MAGICAL_CASING = new SimpleSidedCubeRenderer("casings/magical_casing");
        MANA_HATCH_INPUT_OVERLAY = new SimpleOverlayRenderer("overlay/machine/overlay_mana_input_hatch");
        MANA_HATCH_OUTPUT_OVERLAY = new SimpleOverlayRenderer("overlay/machine/overlay_mana_output_hatch");
        LIFE_ESSENCE_INPUT_OVERLAY = new SimpleOverlayRenderer("overlay/machine/overlay_life_essence_hatch");
        LIFE_ESSENCE_OUTPUT_OVERLAY = new SimpleOverlayRenderer("overlay/machine/overlay_life_essence_hatch");

    }



}

File: src/main/kotlin/rosatech/client/renderer/textures/RosaTextures.kt
package rosatech.client.renderer.textures

import gregtech.client.renderer.texture.cube.SimpleOverlayRenderer

object RosaTextures {
    lateinit var DIMENSIONAL_ENERGY_MULTI: SimpleOverlayRenderer
    lateinit var DIMENSIONAL_ENERGY_4A: SimpleOverlayRenderer
    lateinit var DIMENSIONAL_ENERGY_16A: SimpleOverlayRenderer
    lateinit var DIMENSIONAL_ENERGY_64A: SimpleOverlayRenderer
    lateinit var DIMENSIONAL_ENERGY_256A: SimpleOverlayRenderer
    lateinit var DIMENSIONAL_ENERGY_1024A: SimpleOverlayRenderer

    lateinit var ESSENTIA_INPUT_OVERLAY: SimpleOverlayRenderer
    lateinit var ESSENTIA_OUTPUT_OVERLAY: SimpleOverlayRenderer

    lateinit var BIG_ITEM_FILTER_OVERLAY: SimpleOverlayRenderer

    lateinit var ARCANE_SEALED_CASING: SimpleOverlayRenderer
    lateinit var ELDRITCH_CASING: SimpleOverlayRenderer

    fun preInit() {
        DIMENSIONAL_ENERGY_MULTI = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch")
        DIMENSIONAL_ENERGY_4A = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch_4a")
        DIMENSIONAL_ENERGY_16A = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch_16a")
        DIMENSIONAL_ENERGY_64A = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch_64a")
        DIMENSIONAL_ENERGY_256A = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch_256a")
        DIMENSIONAL_ENERGY_1024A = SimpleOverlayRenderer("overlay/machine/overlay_dimensional_energy_hatch_1024a")

        ESSENTIA_INPUT_OVERLAY = SimpleOverlayRenderer("overlay/machine/overlay_essentia_hatch_input")
        ESSENTIA_OUTPUT_OVERLAY = SimpleOverlayRenderer("overlay/machine/overlay_essentia_hatch_output")

        BIG_ITEM_FILTER_OVERLAY = SimpleOverlayRenderer("overlay/cover/overlay_big_item_filter")

        ARCANE_SEALED_CASING = SimpleOverlayRenderer("casings/thaumcraft/arcane_sealed")
        ELDRITCH_CASING = SimpleOverlayRenderer("casings/thaumcraft/eldritch")
    }
}
